x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportDeleteAutoFollowPatternAction.java

```diff
@@ -74,9 +73,7 @@ static ClusterState innerDelete(DeleteAutoFollowPatternAction.Request request, C
 
         AutoFollowMetadata newAutoFollowMetadata = removePattern(currentAutoFollowMetadata, request.getName());
 
-        return ClusterState.builder(currentState)
-            .metadata(Metadata.builder(currentState.getMetadata()).putCustom(AutoFollowMetadata.TYPE, newAutoFollowMetadata))
-            .build();
+        return currentState.copyAndUpdateMetadata(metadata -> metadata.putCustom(AutoFollowMetadata.TYPE, newAutoFollowMetadata));
```

Comment:
This is sample usage of the new method. This way we do not have to explicitly create a builders for a cluster state and metadata copies, just apply changes on pre-initialized ones.

Reply:
Also as discussed with @original-brownbear we could create even simpler variants for the most common use cases, for example `currentState.copyWithCustomMetadata(AutoFollowMetadata.TYPE, newAutoFollowMetadata);`